cypress/e2e/dashboard.cy.js

```diff
@@ -143,136 +148,176 @@ describe('Validate dashboard page', () => {
-        .children()
-        .each(($item) => {
-          cy.wrap($item).within(() => {
-            cy.get('[data-cy="task-group-list"]')
-              .invoke('text')
-              .then((title) => {
-                expect(title.trim()).to.not.be.empty
-                // check for the sublist of tasks
-                cy.get('[data-cy="taskList"]')
-                  .children('[data-cy="task-link"]')
-                  .should('have.length.gt', 0)
-                cy.get('[data-cy="taskList"]')
-                  .children('[data-cy="task-link"]')
-                  .each(($link) => {
-                    cy.wrap($link).should('not.have.attr', 'href', '#undefined')
-                  })
+    cy.wrap(cardTitles).each((cardTitle) => {
+      // Find the card with the specific title
+      cy.contains('[data-cy="cards"]', cardTitle).as('currentCard')
+      // Check the card title
+      cy.get('@currentCard').should('contain', cardTitle)
+        cy.get('@currentCard').within(() => {
+    cy.get('[data-cy="viewMoreLessButton"]').click()
+    cy.get('[data-cy="task-list"]').should('be.visible')
+    cy.get('[data-cy="task-list"]')
+    .children()
+    .each(($item) => {
+      cy.wrap($item).within(() => {
+        cy.get('[data-cy="task-group-list"]')
+          .invoke('text')
+          .then((title) => {
+            expect(title.trim()).to.not.be.empty
+            // check for the sublist of tasks
+            cy.get('[data-cy="taskList"]')
+              .children('[data-cy="task-link"]')
+              .should('have.length.gt', 0)
+            cy.get('[data-cy="taskList"]')
+              .children('[data-cy="task-link"]')
+              .each(($link) => {
+                cy.wrap($link).should('not.have.attr', 'href', '#undefined')
               })
           })
-        })
+      })
     })
   })
+})
+  })
 
   // EN Tests the Links for Profile page in EI, CPP and OAS but not SIN and Cal
   it('Iterates through EI, CPP and OAS task lists for Profile page EN', () => {
-    // 4 = EI, 5 = CPP and 6 = OAS
-    const EICPPOAS = [4, 5, 6]
-    cy.wrap(EICPPOAS).each((index) => {
-      cy.get(`:nth-child(${index}) > [data-cy="viewMoreLessButton"]`).click()
-      cy.get('[data-cy="most-requested"]').should('be.visible')
-      cy.get(` [data-cy="task-list"]`)
-      cy.get('[data-cy="Task"]')
-      cy.get('[data-cy="task-group-list"]')
-        .parents('[data-cy="Task"]')
-        .find('[data-cy="task-link"] a[href="/en/profile"]')
-        .click({ force: true })
-      cy.location('pathname', { timeout: 10000 }).should('equal', '/en/profile')
-      cy.get('#back-to-dashboard-button').should('exist').click()
-      cy.location('pathname').should('equal', '/en/my-dashboard')
-      cy.get(`:nth-child(${index}) > [data-cy="viewMoreLessButton"]`).click()
-      // cy.url().should('contains', '/en/my-dashboard')
+    const cardTitles = [
+      // 'Canadian Dental Care Plan',
```

Comment:
These should also be tested for instead of being commented out I feel

Reply:
I wasn't sure as it's is just a placeholder.